mlx/ops.cpp

```diff
@@ -1154,11 +1154,45 @@ array allclose(
     const array& b,
     double rtol /* = 1e-5 */,
     double atol /* = 1e-8 */,
+    bool equal_nan /* = false */,
+    StreamOrDevice s /* = {}*/) {
+  return all(isclose(a, b, rtol, atol, equal_nan, s), s);
+}
+
+array isclose(
+    const array& a,
+    const array& b,
+    double rtol /* = 1e-5 */,
+    double atol /* = 1e-8 */,
+    bool equal_nan /* = false */,
     StreamOrDevice s /* = {}*/) {
   // |a - b| <= atol + rtol * |b|
   auto rhs = add(array(atol), multiply(array(rtol), abs(b, s), s), s);
   auto lhs = abs(subtract(a, b, s), s);
-  return all(less_equal(lhs, rhs, s), s);
+  auto out = less_equal(lhs, rhs, s);
+
+  // Correct the result for infinite values.
+  auto any_inf = isinf(a, s) || isinf(b, s) || isneginf(a, s) || isneginf(b, s);
+  auto both_inf =
+      (isinf(a, s) && isinf(b, s)) || (isneginf(a, s) && isneginf(b, s));
+  auto any_nan = isnan(a, s) || isnan(b, s);
+
+  // Convert all elements where either value is infinite to False.
+  out = out && logical_not(any_inf, s);
+
+  // Convert all the elements where both values are infinite and of the same
+  // sign to True.
+  out = out || both_inf;
+
+  // Convert all the elements where either value is NaN to False.
+  out = out && logical_not(any_nan, s);
```

Comment:
I believe this is not needed. If any element is NaN than the `<` operator will be `false` always.

```
>>> 1 <= mx.array(float("nan"))
array(False, dtype=bool)
>>> mx.array(float("nan")) <= 1
array(False, dtype=bool)
```

Reply:
Done.